docs/readme-zephyr.md

```diff
@@ -156,3 +156,56 @@ although MCUboot's key management infrastructure supports multiple keypairs.
 
 Once MCUboot is built, this new keypair file (`mykey.pem` in this
 example) can be used to sign images.
+
+## Serial recovery
+
+### Interface selection
+
+A serial recovery protocol is available over either a serial port or an USB CDC ACM virtual serial port.
+The SMP server implementation can be enabled by ``CONFIG_MCUBOOT_SERIAL=y``.
+To set a type of an interface, use ``BOOT_SERIAL_DEVICE`` Kconfig choice, and select either ``CONFIG_BOOT_SERIAL_UART`` or ``CONFIG_BOOT_SERIAL_CDC_ACM`` value.
+Which interface belongs to the protocol shall be set by the devicetree-chosen node:
+- `zephyr,console` if serial port used.
+- `zephyr,cdc-acm-uart` if virtual serial port used.
+
+### Entering to the serial recovery mode
+
+To enter the serial recovery mode, the device has to initiate rebooting, and a triggering event has to occur, for example, pressing a button.
+
+By default, the serial recovery GPIO pin active state desires entering the mode.
+The pin shall be assigned using the ``mcuboot_button0`` devicetree button alias.
+
+Alternatively, MCUboot can wait for the prescribed duration to see if DFU is invoked by receiving an MCUmgr command.
+Select ``CONFIG_BOOT_SERIAL_WAIT_FOR_DFU=y`` to use this mode. ``CONFIG_BOOT_SERIAL_WAIT_FOR_DFU_TIMEOUT`` option defines
+the amount of time the device will wait for the trigger.
+
+### Direct image upload
+
+The SPM server implementation allows to select image number for DFU when invoking the MCUmgr `image upload` command.
+
+The mapping of image number to partition is as follows:
+* 0 and 1 - image-0, the primary slot of the first image.
+* 2 - image-1, the secondary slot of the first image.
+* 3 - image-2.
+* 4 - image-3.
```

Comment:
Not sure this is right, as far as I know 0 is image 0 slot 0, 1 is image 0 slot 1, 2 is image 1 slot 0, 3 is image 1 slot 1, etc.
Can you double check this?

Reply:
Ping on this comment

Reply:
This is input from @de-nordic. Mapping is not the same as for mcumgr-app. image-2 i image-3 can be reflected as given image slots...